src/cmd/property.go

```diff
@@ -176,42 +223,33 @@ EOF`, getYaml[opslevel.PropertyDefinitionInput]()),
 	},
 }
 
-// The schema in PropertyDefinitionInput can be a nested map[string]any and needs to be handled separately
-func readPropertyDefinitionInput() (*opslevel.PropertyDefinitionInput, error) {
-	d, err := ReadResourceInput[opslevel.JSONSchema](nil)
-	if err != nil {
-		return nil, err
-	}
-	data := *d
-	name, ok := data["name"].(string)
-	if !ok {
-		return nil, fmt.Errorf("name is required and must be a string")
-	}
-	schema, ok := data["schema"].(map[string]any)
-	if !ok {
-		return nil, fmt.Errorf("schema is required and must be a JSON object")
-	}
-	jsonSchema := opslevel.JSONSchema(schema)
-	propDefInput := opslevel.PropertyDefinitionInput{
-		Name:   opslevel.RefOf(name),
-		Schema: &jsonSchema,
+func ReadPropertyAssignInput(input []byte) (*opslevel.PropertyInput, error) {
+	var err error
+	if input == nil {
+		input, err = readInput()
+		if err != nil {
+			return nil, fmt.Errorf("error reading from input: %w", err)
+		}
 	}
+	return ReadResourceHandleJSONFields[opslevel.PropertyInput](input)
+}
 
-	if description, ok := data["description"].(string); ok {
-		propDefInput.Description = opslevel.RefOf(description)
-	}
-	if propertyDisplayStatus, ok := data["propertyDisplayStatus"].(string); ok {
-		propDefInput.PropertyDisplayStatus = opslevel.RefOf(opslevel.PropertyDisplayStatusEnum(propertyDisplayStatus))
+func ReadPropertyDefinitionInput(input []byte) (*opslevel.PropertyDefinitionInput, error) {
```

Comment:
once again the `input` argument is only for testing and so it should not be in this function because its conflating 2 usecases with 1 logical function.  The tests should just use  `ReadResourceHandleJSONFields` and the code in production doesn't have to have `(nil)` everywhere